src/components/admin/Login.jsx

```diff
@@ -1,73 +1,114 @@
-      window.alert('Incorrect password');
-      e.target.password.value = ''; // eslint-disable-line no-param-reassign
-    } else {
-      // Otherwise it is correct and we let the site redirect the client
-      let url = this.props.location.query.url || '';
-      if (url) {
-        _.forEach(this.props.location.query, (q, name) => {
-          if (name !== 'url') {
-            url = `${url}&${name}=${q}`;
+          const auth = window.gapi.auth2.getAuthInstance();
+          if (!auth || !auth.isSignedIn.get()) {
+            window.gapi.auth2.init({
+              client_id: googleClientID,
+            });
+          } else { // detects user has signed in
+            this.onSignInSuccess(auth.currentUser.get());
           }
         });
+        clearInterval(interval);
       }
-      browserHistory.push(url);
+    }, 100);
+  }
+
+  handleRedirect() {
+    // Add additional query params to new url
+    let url = this.props.location.query.url || '';
+    if (url) {
+      _.forEach(this.props.location.query, (q, name) => {
+        if (name !== 'url') {
+          url = `${url}&${name}=${q}`;
+        }
+      });
     }
+    browserHistory.push(url);
+  }
+
+  handleLoginCI() {
+    // if CI, just let it login without interacting with Google Login popup window
+    if (isCI) {
+      this.handleRedirect();
+    }
+  }
+
+  onSignInSuccess(response) {
+    const token = { data: response.getAuthResponse().id_token };
+
+    const xhr = new XMLHttpRequest();
+    xhr.open('POST', '/googlelogin');
+    xhr.setRequestHeader('Content-Type', 'application/json');
+
+    xhr.onreadystatechange = () => {
+      if (xhr.readyState === XMLHttpRequest.DONE && xhr.status === 200) {
+        this.handleRedirect();
+      } else if (xhr.readyState === XMLHttpRequest.DONE) {
+        alert(`Google Login Failed.\n${xhr.status}: ${xhr.statusText}`);
+      }
+    };
+
+    xhr.send(JSON.stringify(token));
+  }
+
+  onSignInFailure(response) {
+    alert(`Google Login Failed.\n${response.error}`);
+  }
+
+  renderButton() {
+    window.gapi.signin2.render('my-signin2', {
+      scope: 'profile email',
+      longtitle: true,
+      onsuccess: this.onSignInSuccess,
+      onfailure: this.onSignInFailure,
+    });
   }
 
   render() {
+    const fade = this.state.GoogleAPIReady ? 1 : 0.5;
     return (
-      <div id="login-page">
-        <form onSubmit={this.handleSubmit}>
-          <TextField
-            name="password"
-            type="password"
-            floatingLabelText="Password"
-            autoFocus
-          /><br /><br />
-          <RaisedButton
-            type="submit"
-            label="Enter Admin Interface"
-            primary
-            style={styles.button}
-          />
-        </form>
+      <div
+        id="my-signin2"
+        style={{
+          ...styles.button,
+          opacity: fade,
+        }}
+        onClick={this.handleLoginCI}
```

Comment:
Hmm, I'm not a huge fan of the fact that the popup window will still open on the CI, but it works, and I guess it doesn't really matter, so if you feel like doing something so that it doesn't happen at all, that would make me happy but otherwise I'll let it slide as it's just semantics.